modules/lib/src/main/resources/assets/js/app/browse/ContentBrowsePanel.ts

```diff
@@ -296,14 +297,51 @@ export class ContentBrowsePanel
         const path: string = this.getPathFromInlinePath(contentInlinePath);
 
         if (path) {
-            // possibly don't need this, but presumes expanding tree structure till the item is found
+            new ContentExistsByPathRequest(path).sendAndParse().then((exists: boolean) => {
+                const targetPath = ContentPath.create().fromString(path).build();
+                this.expandToListElementByPath(this.treeListBox, targetPath.getPathAtLevel(1), targetPath);
+            }).catch(DefaultErrorHandler.handle);
         }
     }
 
     private getPathFromInlinePath(contentPreviewPath: string): string {
         return UriHelper.getPathFromPortalInlineUri(contentPreviewPath, RenderingMode.INLINE);
     }
 
+    private expandToListElementByPath(list: ContentsTreeGridList, itemPath: ContentPath, targetPath: ContentPath): void {
+        let loadCount = 0;
+
+        const itemFinder = () => {
+            const listElement = list.getListElementByPath(itemPath)
+
+            if (listElement) {
+                list.unItemsAdded(itemFinder);
+
+                listElement.whenRendered(() => {
+                    listElement.getHTMLElement().scrollIntoView();
+
+                    if (itemPath.equals(targetPath)) {
```

Comment:
Result of the `itemPath.equals(targetPath)` check should be assigned to a const outside of the loop, as it won't change with iterations.

Reply:
it will be invoked just once, when the search element is finally there